### PR TITLE
Add DurationPicker

### LINKED ISSUE #12: Component: Duration Picker
Displays the field for the duration with some text around it.

### PR DESCRIPTION
**Pull request by panagiotappl**
closes #12

### CODE REVIEW

File: newdle/client/src/components/DurationPicker.module.scss [hunk 1]
```diff
@@ -0,0 +1,13 @@
+@import '../styles/palette.scss';
+
+.duration-picker {
+  background-color: $pink;
+  display: inline-block;
+  padding: 5px;
+  color: white;
```
Comment (mvidalgarcia) [inline newdle/client/src/components/DurationPicker.module.scss]: `$white`

File: newdle/client/src/components/DurationPicker.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import React, {useState} from 'react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+import {Dropdown} from 'semantic-ui-react';
+import styles from './DurationPicker.module.scss';
+
+function renderDurationOptions(min, max, interval) {
```
Comment (ThiefMaster) [inline newdle/client/src/components/DurationPicker.js]: I don't think `render` in the function name makes much sense here

Comment (panagiotappl) [reply newdle/client/src/components/DurationPicker.js]: True. Would a name like `calculate` be more appropriate?

Comment (ThiefMaster) [reply newdle/client/src/components/DurationPicker.js]: not a fan of `calculate` unless math is involved. `getOptions` or `buildOptions` maybe?

Comment (panagiotappl) [reply newdle/client/src/components/DurationPicker.js]: I'll go with `getDurationOptions` then. Sounds better indeed.

File: newdle/client/src/components/DurationPicker.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      />
+      <span> minutes</span>
+    </div>
+  );
+}
+
+DurationPicker.propTypes = {
+  minDuration: PropTypes.number.isRequired,
+  maxDuration: PropTypes.number.isRequired,
+  interval: PropTypes.number.isRequired,
```
Comment (meluru) [inline newdle/client/src/components/DurationPicker.js]: Not a must, but maybe those props could have some default values? Most likely we will use it with the same props everywhere.

Comment (panagiotappl) [reply newdle/client/src/components/DurationPicker.js]: Good idea. Since I'm kind of hardcoding them as I'm passing them from the parent anyway.

File: newdle/client/src/components/DurationPicker.module.scss [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+.duration-picker {
+  background-color: $pink;
+  display: inline-block;
+  padding: 5px;
+  color: $white;
+
+  .dropdown {
+    min-width: 0 !important;
+    border-radius: 0 !important;
```
Comment (pferreir) [inline newdle/client/src/components/DurationPicker.module.scss]: Are you sure there's no alternative to these `!important`?

File: newdle/client/src/components/DurationPicker.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+import {Dropdown} from 'semantic-ui-react';
+import styles from './DurationPicker.module.scss';
+
+function getDurationOptions(min, max, interval) {
+  return _.range(min, max + interval, interval).map(i => ({key: i, value: i, text: i}));
+}
+
+export default function DurationPicker({minDuration, maxDuration, interval}) {
+  const [durationValue, setDurationValue] = useState(minDuration);
+  const durationOptions = getDurationOptions(minDuration, maxDuration, interval);
```
Comment (pferreir) [inline newdle/client/src/components/DurationPicker.js]: This should probably be a fully controlled component. The state is not of much use inside the current component as we will have to use it further into the process.

File: newdle/client/src/components/DurationPicker.module.scss (hunk 4 above)
Comment (panagiotappl) [reply newdle/client/src/components/DurationPicker.module.scss]: Do we need an alternative in this case? It's only class that's going to be used inside the duration-picker.

File: newdle/client/src/components/DurationPicker.js (hunk 5 above)
Comment (ThiefMaster) [reply newdle/client/src/components/DurationPicker.js]: Indeed! IIRC. I didn't comment on that one since until we integrate it with the rest having local state is probably easier.

Comment (panagiotappl) [reply newdle/client/src/components/DurationPicker.js]: The `Timeline` is already merged and since this will be integrated with the `Timeline` I can change it now.

File: newdle/client/src/components/TopHeader.module.scss [hunk 6]
```diff
@@ -1,7 +1,7 @@
 @import '../styles/palette.scss';
 
 .header {
-  background: linear-gradient($coral, $pink);
+  background: $coral;
```
Comment (mvidalgarcia) [inline newdle/client/src/components/TopHeader.module.scss]: Why have you removed the gradient?

File: newdle/core/util.py [hunk 7]
```diff
@@ -1,11 +1,12 @@
 from datetime import datetime
 
-import pytz
 from sqlalchemy import func, types
 from sqlalchemy.sql import operators
 from sqlalchemy.sql.sqltypes import Interval
 from sqlalchemy.util import memoized_property
 
+import pytz
```
Comment (mvidalgarcia) [inline newdle/core/util.py]: What's the reason of this change? I think it was OK.

File: newdle/models.py [hunk 8]
```diff
@@ -1,9 +1,9 @@
-from pytz import timezone, utc
 from sqlalchemy.dialects.postgresql import JSONB
 from sqlalchemy.schema import CheckConstraint
 
 from newdle.core.db import db
 from newdle.core.util import UTCDateTime, format_dt, parse_dt
+from pytz import timezone, utc
```
Comment (mvidalgarcia) [inline newdle/models.py]: And this? `black` messing around?

File: newdle/schemas.py [hunk 9]
```diff
@@ -1,4 +1,5 @@
 from marshmallow import fields, post_dump
+
```
Comment (mvidalgarcia) [inline newdle/schemas.py]: Same.

File: newdle/client/src/components/Timeline.js [hunk 10]
```diff
@@ -143,9 +144,14 @@ export default function Timeline({date, availability, minHour, maxHour, hourStep
...
   return (
     <div className={styles['timeline']}>
-      <Header as="h2" className={styles['timeline-date']}>
-        {moment(date, 'YYYY-MM-DD').format('D MMM YYYY')}
-      </Header>
+      <div className={styles['timeline-title']}>
+        <Header as="h2" className={styles['timeline-date']}>
+          {moment(date, 'YYYY-MM-DD').format('D MMM YYYY')}
+        </Header>
+        <div>
```
Comment (mvidalgarcia) [inline newdle/client/src/components/Timeline.js]: I think you can get rid of this `div`

File: newdle/core/util.py (hunk 7 above)
Comment (ThiefMaster) [reply newdle/core/util.py]: That looks like someone ran isort without the package being properly installed

File: newdle/models.py (hunk 8 above)
Comment (ThiefMaster) [reply newdle/models.py]: same here (isort, not black)
